test/zdtm/static/memfd01.c

```diff
@@ -0,0 +1,112 @@
+#include <err.h>
+#include <fcntl.h>
+#include <linux/memfd.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <sys/vfs.h>
+#include <unistd.h>
+
+#include "zdtmtst.h"
+
+const char *test_doc	= "memfd with different file pointer";
+const char *test_author	= "Nicolas Viennot <Nicolas.Viennot@twosigma.com>";
+
+static int _memfd_create(const char *name, unsigned int flags)
+{
+	return syscall(SYS_memfd_create, name, flags);
+}
+
+int main(int argc, char *argv[])
+{
+	task_waiter_t t;
+	test_init(argc, argv);
+
+	task_waiter_init(&t);
+
+	int fd = _memfd_create("somename", MFD_CLOEXEC);
+	if (fd < 0)
+		err(1, "Can't call memfd_create");
+
+	pid_t pid = getpid();
```

Comment:
According to the Linux kernel coding styles (we use it for criu), all variables have to be declared at the start of scope.